src/DataLoader.Uno/DataLoaderViewController.cs

```diff
@@ -0,0 +1,407 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Chinook.DataLoader
+{
+	/// <summary>
+	/// This controller encapsulates all the reusable logic needed by <see cref="DataLoaderView"/> or other view implementations of <see cref="IDataLoader"/>.<br/>
+	/// It limits the view updates of the associated <see cref="IDataLoader"/> using <see cref="StateChangingThrottleDelay"/> and <see cref="StateMinimumDuration"/>.
+	/// </summary>
+	public class DataLoaderViewController
+	{
+		// Data states
+		public const string DataVisualGroup = "DataStates";
+		public const string InitialVisualStateName = "Initial";
+		public const string DataVisualStateName = "Data";
+		public const string EmptyVisualStateName = "Empty";
+
+		// Loading states
+		public const string LoadingVisualGroup = "LoadingStates";
+		public const string LoadingVisualStateName = "Loading";
+		public const string NotLoadingVisualStateName = "NotLoading";
+
+		// Error states
+		public const string ErrorVisualGroup = "ErrorStates";
+		public const string ErrorVisualStateName = "Error";
+		public const string NoErrorVisualStateName = "NoError";
+
+		// Combined states
+		public const string CombinedVisualGroup = "CombinedStates";
+
+		private readonly IDataLoaderViewDelegate _delegate;
```

Comment:
Dumb question, could it be a good idea to have a weakreference here, since we could potentially have a circular reference?

Reply:
good point!